tests/unit_tests/test_elasticsearchstore.py

```diff
@@ -301,3 +255,63 @@ def test__update_latest_metric_if_necessary(test_elastic_metric, test_elastic_la
                                             create_store):
     create_store._update_latest_metric_if_necessary(test_elastic_metric, run)
     assert run.latest_metrics == test_elastic_latest_metrics
+
+
+@pytest.mark.parametrize("test_parsed_filter,test_query,test_type",
+                         [({'type': 'parameter', 'key': 'param0',
+                            'comparator': 'LIKE', 'value': '%va%'},
+                           Q("term", params__key="param0") &
+                           Q('bool', must=[Q("wildcard", params__value="*va*")]),
+                           "params"),
+                          ({'type': 'parameter', 'key': 'param0',
+                            'comparator': 'ILIKE', 'value': '%va%'},
+                           Q("term", params__key="param0") &
+                           Q('bool', must=[Q("wildcard", params__value="*va*")]),
+                           "params"),
+                          ({'type': 'parameter', 'key': 'param0',
+                            'comparator': '=', 'value': 'va'},
+                           Q("term", params__key="param0") &
+                           Q('bool', must=[Q("term", params__value="va")]),
+                           "params"),
+                          ({'type': 'metric', 'key': 'metric0', 'comparator': '>', 'value': '1'},
+                           Q("term", latest_metrics__key="metric0") &
+                           Q('bool', must=[Q("range", latest_metrics__value={'gt': "1"})]),
+                           "latest_metrics"),
+                          ({'type': 'metric', 'key': 'metric0', 'comparator': '>=', 'value': '1'},
+                           Q("term", latest_metrics__key="metric0") &
+                           Q('bool', must=[Q("range", latest_metrics__value={'gte': "1"})]),
+                           "latest_metrics"),
+                          ({'type': 'metric', 'key': 'metric0', 'comparator': '<', 'value': '1'},
+                           Q("term", latest_metrics__key="metric0") &
+                           Q('bool', must=[Q("range", latest_metrics__value={'lt': "1"})]),
+                           "latest_metrics"),
+                          ({'type': 'metric', 'key': 'metric0', 'comparator': '<=', 'value': '1'},
+                           Q("term", latest_metrics__key="metric0") &
+                           Q('bool', must=[Q("range", latest_metrics__value={'lte': "1"})]),
+                           "latest_metrics"),
+                          ({'type': 'tag', 'key': 'tag0', 'comparator': '!=', 'value': 'val2'},
+                           Q("term", tags__key="tag0") &
+                           Q('bool', must_not=[Q("term", tags__value="val2")]),
+                           "tags")])
+@pytest.mark.usefixtures('create_store')
+def test___build_elasticsearch_query(test_parsed_filter, test_query,
+                                     test_type, create_store):
+    actual_s = create_store._build_elasticsearch_query(
```

Comment:
please give an explicit name to the variable, like actual_query or something like that